core/variant.cpp

```diff
@@ -1582,6 +1582,13 @@ Variant::operator String() const {
 					};
 				};
 #endif
+
+				ScriptInstance *script_instance = _get_obj().obj->get_script_instance();
+				if (script_instance) {
+					String ret = script_instance->to_string();
+					if (ret != String())
+						return ret;
```

Comment:
I don't think it's a good idea to use an empty string to determine whether the script could do the conversion. Instead I would make it `String ScriptInstance::to_string(bool &r_valid)` or `bool ScriptInstance::to_string(String &r_out)`.